cranelift/interpreter/src/frame.rs

```diff
@@ -66,21 +71,39 @@ impl<'a> Frame<'a> {
     pub fn rename(&mut self, old_names: &[ValueRef], new_names: &[ValueRef]) {
         trace!("Renaming {:?} -> {:?}", old_names, new_names);
         assert_eq!(old_names.len(), new_names.len());
-        let mut registers = HashMap::with_capacity(self.registers.len());
-        for (on, nn) in old_names.iter().zip(new_names) {
-            let v = self.registers.get(on).unwrap().clone();
-            registers.insert(*nn, v);
+        let new_registers = vec![None; self.registers.len()];
+        let mut old_registers = std::mem::replace(&mut self.registers, new_registers);
+        self.registers = vec![None; self.registers.len()];
+        for (&on, &nn) in old_names.iter().zip(new_names) {
+            let value = std::mem::replace(&mut old_registers[on.index()], None);
+            self.registers[nn.index()] = value;
         }
-        self.registers = registers;
+    }
+
+    /// Accessor for the current entries in the frame.
+    pub fn entries_mut(&mut self) -> &mut [Option<DataValue>] {
+        &mut self.registers
     }
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
     use cranelift_codegen::data_value::DataValue;
+    use cranelift_codegen::ir::immediates::{Ieee32, Ieee64};
     use cranelift_codegen::ir::InstBuilder;
     use cranelift_frontend::{FunctionBuilder, FunctionBuilderContext};
+    use cranelift_reader::parse_functions;
+
+    /// Helper to create a function from CLIF IR.
+    fn function(code: &str) -> Function {
+        parse_functions(code).unwrap().into_iter().next().unwrap()
+    }
+
+    /// Helper to create a frame for a CLIF function.
+    fn frame(function: &Function) -> Frame {
+        Frame::new(&function)
+    }
```

Comment:
Is this helper carrying its weight? It only saves five characters and doesn't *do* anything different (like unwrap or provide default values, etc).

Reply:
Gone!